lib/fs.js

```diff
@@ -376,52 +263,30 @@ function readFileAfterStat(err, stats) {
   context.read();
 }
 
-function readFileAfterRead(err, bytesRead) {
-  var context = this.context;
-
-  if (err)
-    return context.close(err);
-
-  if (bytesRead === 0)
-    return context.close();
-
-  context.pos += bytesRead;
-
-  if (context.size !== 0) {
-    if (context.pos === context.size)
-      context.close();
-    else
-      context.read();
-  } else {
-    // unknown size, just read until we don't get bytes.
-    context.buffers.push(context.buffer.slice(0, bytesRead));
-    context.read();
-  }
-}
-
-function readFileAfterClose(err) {
-  var context = this.context;
-  var buffer = null;
-  var callback = context.callback;
-
-  if (context.err || err)
-    return callback(context.err || err);
+function readFile(path, options, callback) {
+  callback = maybeCallback(callback || options);
+  options = getOptions(options, { flag: 'r' });
+  if (!ReadFileContext)
+    ReadFileContext = require('internal/fs/readfilecontext');
```

Comment:
I believe we use underscores in file names, can you rename this to `read_file_context`?

Reply:
Given that the object name is `ReadFileContext` without any separation, I'd prefer not to, but can.